juju/juju
# Fix build tags being overridden and add build tag reflection in version info.

Jenkins jobs that build juju binaries have been passing an empty BUILD_TAGS variable, but this disables dqlite by excluding the dqlite build tag. Instead we append default tags (dqlite and sqlite3) to the list passed in by the script/user.

Since the build tags are now always set, we introduce a negating build tag, `no-dqlite`, to disable adding in the default dqlite build tags.

## QA steps

- Build juju with only lxd provider
- `make install BUILD_TAGS="minimal provider_lxd"`
- Bootstrap lxd and check it is working (i.e. should have dqlite)
- Check `juju version --all` has all build flags
```
juju version --all
version: 3.2.2-ubuntu-amd64
git-commit: 5316fe0e94d6d6237a864da03b2349a8b91e382e
git-tree-state: clean
compiler: gc
go-build-tags: minimal,provider_lxd,libsqlite3,dqlite
```

## Documentation changes

N/A

## Bug reference

https://jenkins.juju.canonical.com/job/build-juju/12887/consoleText
